services/src/test/java/org/exoplatform/addons/gamification/rest/TestChallengeRest.java

```diff
@@ -409,11 +416,37 @@ public void testGetChallengesByUser() {
             assertEquals(0, savedChallenges.size());
 
             startSessionAs("root1");
+
+            restPath = "/gamification/challenge/api/allChallenge?offset=0&limit=10&domainId="+domain.getId()+"&announcements=4";
+            httpRequest = new MockHttpServletRequest(restPath, null, 0, "GET", null);
+            envctx.put(HttpServletRequest.class, httpRequest);
+            response = launcher.service("GET", restPath, "", h, null, envctx);
+            assertNotNull(response);
+            assertEquals(200, response.getStatus());
+            savedChallenges = (List<ChallengeRestEntity>) response.getEntity();
+            assertEquals(2, savedChallenges.size());
+
+            restPath = "/gamification/challenge/api/allChallenge?offset=0&limit=10&domainId=0&announcements=4&groupByDomain=true";
+            httpRequest = new MockHttpServletRequest(restPath, null, 0, "GET", null);
+            envctx.put(HttpServletRequest.class, httpRequest);
             response = launcher.service("GET", restPath, "", h, null, envctx);
             assertNotNull(response);
             assertEquals(200, response.getStatus());
             savedChallenges = (List<ChallengeRestEntity>) response.getEntity();
             assertEquals(1, savedChallenges.size());
+
+            restPath = "/gamification/challenge/api/allChallenge?offset=0&limit=10";
+            httpRequest = new MockHttpServletRequest(restPath, null, 0, "GET", null);
+            envctx.put(HttpServletRequest.class, httpRequest);
+            response = launcher.service("GET", restPath, "", h, null, envctx);
+            assertNotNull(response);
+            assertEquals(200, response.getStatus());
+            savedChallenges = (List<ChallengeRestEntity>) response.getEntity();
+            assertEquals(2, savedChallenges.size());
+
+
+
+
         } catch (Exception e) {
```

Comment:
```suggestion
```
Having this kind of catch will make the test never fail, please delete it (I know it's an old code, but we need to have it as a real Unit Test. Thanks)